src/ServiceControl.Monitoring/Http/Diagrams/TimingsAggregator.cs

```diff
@@ -21,22 +25,35 @@ public IEnumerable<MonitoredEndpoint> AggregateIntoLogicalEndpoints()
 
             var processingTimes = processingTimeStore.GetTimings(now);
             var criticalTimes = criticalTimeStore.GetTimings(now);
+            var queueLengths = queueLengthDataStore.GetQueueLengths(now);
 
-            var endpointNames = processingTimes.Concat(criticalTimes).Select(t => t.Id.EndpointName).Distinct();
+            var endpointNames = processingTimes.Concat(criticalTimes).Select(t => t.Id.EndpointName).Concat(queueLengths.Keys).Distinct();
 
             foreach (var endpointName in endpointNames)
             {
                 var instanceProcessingTime = processingTimes.Where(pt => pt.Id.EndpointName == endpointName).ToList();
                 var instanceCriticalTime = criticalTimes.Where(ct => ct.Id.EndpointName == endpointName).ToList();
+                var queueLength = queueLengths.FirstOrDefault(ql => ql.Key == endpointName);
```

Comment:
_This comment is not only for this PR. Feel free to mark it as a separate issue._

I've just noticed that we're scanning over the collections to search for a specific item. This means complexity of O (n^2). If that's the case, finding by the endpoint name, should find better structures (dicts) to find these?